pkg/gatewaysecret/secret.go

```diff
@@ -0,0 +1,61 @@
+package gatewaysecret
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apicorev1 "k8s.io/api/core/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	LastModifiedAtAnnotation = "lastModifiedAt"
+)
+
+type secretManager struct {
+	kcpClient client.Client
+}
+
+func (sm *secretManager) findSecret(ctx context.Context, objKey client.ObjectKey) (*apicorev1.Secret, error) {
+	secret := &apicorev1.Secret{}
+
+	err := sm.kcpClient.Get(ctx, objKey, secret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get secret %s: %w", objKey.Name, err)
+	}
+
+	return secret, nil
+}
+
+// isNotFound returns true if the error is a NotFound error.
+func isNotFound(err error) bool {
```

Comment:
we have utility isNotFound https://github.com/kyma-project/lifecycle-manager/blob/0d91106f187024f74f65846cf78222d8ad471079/pkg/util/error.go#L15

reuse that please